add airflow troubleshooting dag documentation (#35470)
DataDog/documentation: content/en/data_observability/jobs_monitoring/airflow_troubleshooting_dag.md

---
title: "Troubleshoot Airflow Setup with the OpenLineage Validation DAG"
description: "Use the OpenLineage validation DAG to diagnose configuration issues with Airflow Data Observability: Jobs Monitoring."
further_reading:
  - link: '/data_observability/jobs_monitoring/airflow/'
    tag: 'Documentation'
    text: 'Enable Data Observability: Jobs Monitoring for Apache Airflow'
  - link: '/data_jobs'
    tag: 'Documentation'
    text: 'Data Observability: Jobs Monitoring'
---

## Overview

The OpenLineage validation DAG validates your Airflow OpenLineage configuration and reports any issues that would prevent Data Observability: Jobs Monitoring from receiving events.

Run this DAG when:

> - Airflow jobs do not appear in the [Data Observability: Jobs Monitoring][1] page after completing setup.

Reviewer: ```suggestion
- Airflow jobs do not appear on the [Data Observability: Jobs Monitoring][1] page after you complete setup.
```

- You want to verify your configuration before running production workloads.

The DAG checks the following:

| Check | What it validates |
|---|---|
| Package installation | OpenLineage package is installed and importable |
| Package version | Installed version compared to the latest available on PyPI |
| Listener accessibility | OpenLineage plugin listener can be loaded by Airflow |
| Enabled state | OpenLineage is not disabled by environment variable or Airflow config |
| Transport configuration | A valid transport is configured (HTTP or Datadog) |
| Datadog endpoint | Transport URL points to a Datadog intake endpoint |
| Network connectivity | TCP connection to the configured backend URL succeeds |
| Configuration conflicts | No duplicate or conflicting transport or config file settings |

## Run the OpenLineage validation DAG

### Prerequisites

- Access to your Airflow DAGs directory (local path or cloud storage bucket for MWAA)

> - Airflow scheduler and worker pods or processes must be running

Reviewer: ```suggestion
- Access to your Airflow DAGs directory (a local path, or a cloud storage bucket for Amazon MWAA)
- A running Airflow scheduler and workers (pods or processes)
```

> ### Step 1: Add the DAG file

Reviewer: ```suggestion
### Add the DAG file
```

        log.warning(conflict_msg)
        conflicts.append(conflict_msg)

    config_files = []

    config_path_env = os.getenv("OPENLINEAGE_CONFIG")
    if config_path_env and os.path.exists(os.path.expanduser(config_path_env)):
        config_files.append(f"OPENLINEAGE_CONFIG={config_path_env}")

    airflow_config_path = os.getenv("AIRFLOW__OPENLINEAGE__CONFIG")
    if airflow_config_path and os.path.exists(os.path.expanduser(airflow_config_path)):
        config_files.append(f"AIRFLOW__OPENLINEAGE__CONFIG={airflow_config_path}")

    if os.path.exists("openlineage.yml"):
        config_files.append("openlineage.yml in current directory")

    if os.path.exists(os.path.expanduser("~/.openlineage/openlineage.yml")):
        config_files.append("~/.openlineage/openlineage.yml")

    if len(config_files) > 1:
        conflict_msg = "Multiple config files found: " + ", ".join(config_files)
        log.warning(conflict_msg)
        conflicts.append(conflict_msg)

    if not os.getenv("OPENLINEAGE_NAMESPACE"):
        log.info("OPENLINEAGE_NAMESPACE is not set. Default namespace will be used.")

    validation_results["conflicts"] = conflicts
    return conflicts
```

> ### Step 2: Trigger the DAG

Reviewer: ```suggestion
### Trigger the DAG
```

After the Airflow scheduler picks up the new file, trigger the DAG manually:

1. In the Airflow UI, navigate to **DAGs** and find `openlineage_preflight_check_dag`.
2. Toggle the DAG on if it is paused.

> 3. Click the **Trigger DAG** button (play icon).

Reviewer: ```suggestion
3. Click **Trigger DAG** (play icon).
```

> ### Step 3: View the logs

Reviewer: ```suggestion
### View the logs
```

1. In the Airflow UI, navigate to **DAGs > openlineage_preflight_check_dag**.
2. Click the most recent DAG run.
3. Click the **validate_setup** task.
4. Select the **Logs** tab.

> ## Read the output

Reviewer: ```suggestion
## Understand the output
```

The DAG logs two sections: an environment summary printed at the start, and a validation summary printed at the end.

### Environment information

The first section shows what the DAG detected about your environment:

```
===== Environment Information =====
Airflow Version: 2.9.2
OpenLineage Python Version: 1.24.2
OpenLineage Provider Version: 1.14.0 (apache-airflow-providers-openlineage)
This Airflow installation is running on Amazon MWAA
===================================
```

### Validation summary

> The validation summary appears at the end of the task log. Each line starts with `✓`, `✗`, or `!`:

Reviewer: ```suggestion
The validation summary appears at the end of the task log. Each line starts with one of three symbols:
```

- `✓` — check passed
- `✗` — check failed (events are not sent to Datadog)

> - `!` — warning (may indicate a configuration issue)

Reviewer: ```suggestion
- `✓`: Check passed.
- `✗`: Check failed (events are not sent to Datadog).
- `!`: Warning (may indicate a configuration issue).
```

> A healthy setup looks like this:

Reviewer: ```suggestion
The following output indicates a healthy setup:
```

```
===== OpenLineage Validation Summary =====
✓ OpenLineage Package: apache-airflow-providers-openlineage version 2.7.3
✓ OpenLineage is enabled
✓ OpenLineage listener is accessible
✓ Transport Type: Datadog
✓ Integration: Datadog
✓ Network connectivity to backend is successful
✓ No configuration conflicts detected
========================================
OpenLineage appears to be configured properly, but check the logs for warnings
```

### Interpreting failures

Use this table to resolve common failures:

> | Log message | Cause | Resolution |

Reviewer: Please add periods to all of the descriptions in this table to make it consistent.

| `✗ Transport Type: Console (won't send events to Datadog)` | Transport is set to `console` | Change the transport to `datadog` or `http` pointing to the Datadog intake URL |
| `✗ Network connectivity check failed` | Airflow workers cannot reach the Datadog intake endpoint | Check firewall or network policies; confirm the URL and port 443 are accessible |
| `! Configuration conflicts detected` | Multiple transport or config file sources are active | Remove duplicate configurations and keep one authoritative source |

**Note**: If `✗ OpenLineage listener is not accessible` appears together with package installation failures, the package is likely installed only on the scheduler and not on the workers. OpenLineage requires the provider on both.

### Version warnings

If the DAG reports a version warning such as:

```
`apache-airflow-providers-openlineage` is not the newest version.
Installed version: `1.8.0`, Latest version: `2.7.3`
```

Consider upgrading to the latest version to benefit from bug fixes and improved Datadog compatibility. For Amazon MWAA, see [Upgrade OpenLineage provider on Amazon MWAA][3] for version-specific instructions.

> ## Further Reading

Reviewer: ```suggestion
## Further reading
```
